src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,160 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            addFirst(value);
+        } else {
+            addLast(value);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkAddIndex(index);
+        if (head == null) {
+            addFirst(value);
+        } else if (size == index) {
+            addLast(value);
+        } else {
+            Node<T> currentIndexNode = getNodeByIndex(index);
+            Node<T> newNode = new Node<>(currentIndexNode.prev, value, currentIndexNode);
+            if (currentIndexNode.prev != null) {
+                currentIndexNode.prev.next = newNode;
+            } else {
+                head = newNode;
+            }
+            currentIndexNode.prev = newNode;
+        }
+        size++;
+    }
+
+    private void addFirst(T value) {
+        Node<T> nodeToAdd = new Node<>(null, value, null);
+        head = nodeToAdd;
+        tail = nodeToAdd;
+    }
+
+    private void addLast(T value) {
+        Node<T> nodeToAdd = new Node<>(tail, value, null);
+        tail.next = nodeToAdd;
+        tail = nodeToAdd;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
```

Comment:
The addAll method does not check if the provided list is null. This could lead to a NullPointerException if a null list is passed. Consider adding a null check for the list parameter.